src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -187,7 +188,12 @@ string EnsureRequiredTagIsPresent(DicomTag dicomTag)
             {
```

Comment:
Probably not your change. But can we do this logic outside the exception? Exceptions are to unwind the stack all the way up. If there are expected exceptions, they should be handled differently like a bool return with TryValidate() and do the necessary action